velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -41,6 +30,78 @@ void ensureRegexIsConstant(
   }
 }
 
+FOLLY_ALWAYS_INLINE std::string prepareRegexpPattern(
+    const StringView& pattern) {
+  static const RE2 kRegex("[(][?]<([^>]*)>");
+
+  // (?<name>regex) => (?P<name>regex)
+  std::string newPattern = pattern.getString();
+  RE2::GlobalReplace(&newPattern, kRegex, R"((?P<\1>)");
+
+  return newPattern;
+}
+
+FOLLY_ALWAYS_INLINE std::string prepareReplacement(
+    const RE2& re,
+    const StringView& replacement) {
+  if (replacement.size() == 0) {
+    return std::string{};
+  }
+
+  auto newReplacement = replacement.getString();
+
+  // If newReplacement contains a reference to a
+  // named capturing group ${name}, replace the name with its index.
+  static const RE2 kExtractRegex(R"(\${([^}]*)})");
+  VELOX_DCHECK(
+      kExtractRegex.ok(),
+      "Invalid regular expression {}: {}.",
+      R"(\${([^}]*)})",
```

Comment:
Could be put after 'Invalid regular expression' in the error message?

Reply:
Sorry, I didn't catch that. Could you explain it again?

Reply:
@rui-mo